src/Blazored.Modal/Services/IModalService.cs

```diff
@@ -1,40 +1,36 @@
 ﻿using System;
+using System.Collections.Generic;
 using Microsoft.AspNetCore.Components;
 
 namespace Blazored.Modal.Services
 {
     public interface IModalService
     {
-        /// <summary>
-        /// Invoked when the modal component closes.
-        /// </summary>
-        event Action<ModalResult> OnClose;
-
         /// <summary>
         /// Shows the modal with the component type.
         /// </summary>
-        void Show<T>() where T : ComponentBase;
+        ModalReference Show<T>() where T : ComponentBase;
 
         /// <summary>
         /// Shows the modal using the specified title and component type.
         /// </summary>
         /// <param name="title">Modal title.</param>
-        void Show<T>(string title) where T : ComponentBase;
+        ModalReference Show<T>(string title) where T : ComponentBase;
 
         /// <summary>
         /// Shows the modal using the specified title and component type.
         /// </summary>
         /// <param name="title">Modal title.</param>
         /// <param name="options">Options to configure the modal.</param>
-        void Show<T>(string title, ModalOptions options) where T : ComponentBase;
+        ModalReference Show<T>(string title, ModalOptions options) where T : ComponentBase;

```

Comment:
Maybe make the comments on these also use `<paramref>`, and also mention ModalOptions on the overload with ModalOptions